network-area-diagram/src/main/java/com/powsybl/nad/build/iidm/VoltageLevelFilter.java

```diff
@@ -71,6 +73,14 @@ public static VoltageLevelFilter createVoltageLevelsFilter(Network network, List
         return createVoltageLevelsDepthFilter(network, voltageLevelIds, 0);
     }
 
+    public static Collection<VoltageLevel> getNextDepthVoltageLevels(Network network, List<VoltageLevel> voltageLevels) {
+        List<String> voltageLevelIds = voltageLevels.stream().map(VoltageLevel::getId).collect(Collectors.toList());
+        VoltageLevelFilter voltageLevelFilter = createVoltageLevelsDepthFilter(network, voltageLevelIds, 1);
+        Set<VoltageLevel> voltageLevelSet = new HashSet<>(voltageLevelFilter.getVoltageLevels());
+        voltageLevels.forEach(voltageLevelSet::remove);
```

Comment:
I think we can do better to avoid creating those sets. Maybe by using `traverseVoltageLevels` at the cost of changing its signature. Or maybe by enriching VoltageLevelFilter API, like in #491. But let's do that later in a separate PR